### PR TITLE
Association

### PR DESCRIPTION
**Pull request by Gbengacode**
### I made the following changes to this work

- Create the has-many/belongs-to relationship between Classroom and Student
- Create a class Book
- Create a class Rental
- Create the many-to-many relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require_relative('./rental')
+class Book
+  attr_accessor :title, :author
+  attr_reader :rental
```
Comment (TedLivist) [inline book.rb]: - [ ] The getter and setter (`attr_accessor`) for `:rental` should be present, not `attr_reader`. Kindly fix this.

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
+
+  def initialize(date)
+    @date = date
+    @person = person
+    @book = book
+    person.rental << self
+    book.rental << self
```
Comment (TedLivist) [inline rental.rb]: - [ ] There should be `person` and `book` parameters in your **`initialize`** method, for your respective instance variables to work. Kindly add them.

  Remember to ensure your `Rental.new` return statements in the `add_rental` methods in both **Person** and **Book** classes follow the same declarations.

File: person.rb [hunk 3]
```diff
@@ -1,14 +1,16 @@
 require_relative('./corrector')
+require_relative('./rental')
 class Person
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rental
```
Comment (TedLivist) [inline person.rb]: - [ ] `:rental` should have getter and setter not getter only. Kindly fix this.

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (TedLivist) [inline rental.rb]: - [ ] `:book` and `:person` should have getters and setters (`attr_accessor`) not only getters (`attr_reader`). Kindly fix this.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Gbengacode,

Good job so far!
✔️ Good documentation
✔️ Establishment of association between classroom and student classes
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Gbenga, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**